### PR TITLE
Use YAML metadata and HTML5.

### PR DESCRIPTION
**Pull request by adunning**
This allows the author's name etc. to be defined only once in the metadata, allowing it to appear in both the HTML and LaTeX templates (and improving the DOCX output). It's tempting to set `abstract` and `acknowledgements` as variables as well (this fits better with the particular class I'm using, [ut-thesis](https://www.ctan.org/tex-archive/macros/latex/contrib/ut-thesis?lang=en)), but this would make it more complicated to have them show up in the DOCX and the HTML table of contents.

I've also added a few lines to the CSS to stop superscripts from disturbing the line spacing when footnotes are added.

Since the `template.tex` file has already been modified from the Pandoc default, I found that removing the separate `preamble.tex` simplified matters. Format-specific variables have been moved to the `makefile` wherever possible.

The example references file has been updated to use CSL JSON format (native to Pandoc) rather than BibTeX.

### CODE REVIEW

**Comment by tompollard** [comment]
Thanks for submitting these improvements @adunning.

It looks like the PDF build is failing on Travis, though I'm not immediately sure why:
https://travis-ci.org/tompollard/phd_thesis_markdown/builds/78093404

Running pandoc in verbose mode (adding the --verbose flag to the Makefile) usually identifies the issue. I'll have a look this evening, unless you get a chance to deal with it in the meantime :)

**Comment by adunning** [comment]
The build is fine on my own system; I imagine that there's probably some required LaTeX package that isn't being installed.

File: style/template.tex [hunk 1]
```diff
@@ -60,15 +53,42 @@
...
 \usepackage[$for(geometry)$$geometry$$sep$,$endfor$]{geometry}
 $endif$
+\ifxetex
+  \usepackage[setpagesize=false, % page size defined by xetex
+              unicode=false, % unicode breaks when used with xetex
+              xetex]{hyperref}
+\else
+  \usepackage[unicode=true]{hyperref}
+\fi
+\usepackage[usenames,dvipsnames]{color}
```
Comment (adunning) [inline style/template.tex]: I think this is what's causing the build issue; it's from the default Pandoc template. I haven't yet figured out why this is a problem, given that it works locally.

**Comment by adunning** [comment]
I can only guess that it has something to do with the fact that the server is running TeX Live 2012, while locally I have TeX Live 2015.

**Comment by adunning** [comment]
I've made adjustments to the template allowing it to be used with LaTeX 2012.

**Comment by tompollard** [comment]
Great, thanks, and sorry for not getting back to you on this earlier. I'll take a look shortly!

**Comment by adunning** [comment]
Have you had a chance to look at this? Happy to make any further changes you deem necessary.

**Comment by tompollard** [comment]
Sorry not looked yet, I've been totally swamped with work over the past couple of weeks. Will look shortly!

**Comment by tompollard** [comment]
Hey Andrew, sorry again that it's taken me so long to look at this. The changes are all work well, so I'm tempted to merge...but I guess my question is whether it's a good idea to drop bib in favour of json? 

I've just checked my reference manager and it only appears to export to BibTeX (*.bib), Research Information Systems (*.ris), and EndNote XML (*.xml).

My hope was to make the build process simple for non-techies, and I'm worried that moving to json would increase the barrier to entry. Ideally we would make it simple for the author to choose between formats...

**Comment by adunning** [comment]
Thanks for looking at this! That is a good thought: Pandoc-Citeproc can read any of those, but Citeproc JSON is its native format, so it's ultimately the most straightforward. While most reference managers theoretically offer a BibTeX export option, the results tend to be idiosyncratic (usually due to poor correlation between the reference manager's fields and those in BibTeX) and cannot be parsed consistently. After testing many different options, I found that the most user-friendly setup is to use Zotero with [Better BibTeX](https://zotplus.github.io/better-bibtex/), which provides a specific Pandoc JSON option and allows for customizable citation keys. My thinking is that any user technical enough to be using a true BibTeX manager such as BibDesk, JabRef, or the like will also be able to change the file extension in the makefile.

**Comment by tompollard** [comment]
(not forgotten about this Andrew...just need to spend a bit of time looking over everything before I decide how to merge...)

**Comment by adunning** [comment]
Great; I can squish these down into one commit if desired, as it has become as bit messy as I keep it updated.

**Comment by thomaseizinger** [comment]
Hi,
i am interested in these changes. Are they going to be merged soon?

**Comment by adunning** [comment]
I had entirely forgotten about this. This branch is now quite outdated, and I wouldn't recommend using it.

**Comment by adunning** [comment]
A new Pandoc thesis template should use the pandoc-crossref filter. My thesis ended up diverging significantly from this to meet the University of Toronto's specific requirement, so it's non-trivial to update it, but I may put up a version of it if there is demand.

**Comment by thomaseizinger** [comment]
If you find time to share it, I would greatly appreciate it!

**Pull request closed without merging**